src/translator/service.h

```diff
@@ -6,34 +6,86 @@
 #include <vector>
 
 #include "data/types.h"
+#include "logging.h"
 #include "quality_estimator.h"
 #include "response.h"
 #include "response_builder.h"
+#include "tensors/tensor_allocator.h"
 #include "text_processor.h"
 #include "threadsafe_batching_pool.h"
 #include "translation_model.h"
 #include "translator/parser.h"
 #include "vocabs.h"
-
 namespace marian {
 namespace bergamot {
 
+class Workspace {
+ public:
+  Workspace(size_t deviceId, size_t workspaceSizeInMB)
+      : device_(deviceId, DeviceType::cpu), precision_(typeFromString("float32")) {
+    // We'll eventually get rid of this, but proof of concept faster?
+    auto graph = New<ExpressionGraph>(/*inference=*/true);  // set the graph to be inference only
+    graph->setDefaultElementType(precision_);
+    graph->setDevice(device_);
+
+    // TODO(@jerinphilip): Are we just fine without configuring? Does Backend have sane defaults?
+    // This however does not affect memory allocations.
+    graph->getBackend()->configureDevice(horribleOptionsHack());
+
+    Ptr<Backend> backend = graph->getBackend();
+
+    tensors_ = New<TensorAllocator>(backend);
+    tensors_->reserve(workspaceSizeInMB);
+
+    cache_ = New<TensorAllocator>(backend);
+  }
+
+  Ptr<TensorAllocator> tensors() { return tensors_; }
+  Ptr<TensorAllocator> cache() { return cache_; }
+
+  size_t id() const { return device_.no; }
+  marian::DeviceId device() const { return device_; }
+  marian::Type precision() const { return precision_; }
+
+  void clear() {
+    tensors_->clear();
+    cache_->clear();
+  }
```

Comment:
We clear tensors_ and cache_ (there's longterm and short-term memory which just stores hash Id, refs into tensors_ or cache_ probably tensors).